src/framework/level1.js

```diff
@@ -2,30 +2,24 @@
 
 const {GObject, Gtk} = imports.gi;
 const {logoIDToResource, LogoImage, VALID_LOGOS} = imports.framework.logoImage;
+const {PopupMenu} = imports.popupMenu;
 
 var FrameworkLevel1 = GObject.registerClass({
     GTypeName: 'FrameworkLevel1',
     Template: 'resource:///com/endlessm/HackToolbox/framework/level1.ui',
     InternalChildren: ['accentColorButton', 'fontButton', 'fontSizeAdjustment',
-        'infoColorButton', 'logoButton', 'logoChoices', 'logoColorButton',
-        'logoMenu', 'mainColorButton'],
+        'infoColorButton', 'logoButton', 'logoColorButton', 'mainColorButton'],
 }, class FrameworkLevel1 extends Gtk.Grid {
     _init(props = {}) {
         super._init(props);
 
+        const logoChoices = {};
```

Comment:
I wonder if it wouldn't be easier to instead have this as a `Map`. That way, if you stored it in the menu instead of separately storing the keys and values, you could potentially avoid the double scan in `_updateForPropValue()`

Reply:
Fixed in `popupMenu`.